(feat): Agent Control (#19943)
newrelic/docs-website: src/content/docs/new-relic-control/agent-control/troubleshooting.mdx

Reviewer: @alvarocabanas can you check that the details you added to the preview docs were not lost?

Author: I'll double-check and add any troubleshooting content that went in to the LP docs recently.

Reviewer: Thanks!

> ---

Reviewer: I added in the preview docs this section that I think it makes sense to keep https://github.com/newrelic/beta-docs-site/pull/478/files

title: "Troubleshoot common issues"
metaDescription: "Learn how to diagnose and resolve common problems with Agent Control setup."
freshnessValidatedDate: never
---

<Callout title="preview">
  We're still working on this feature, but we'd love for you to try it out!

  This feature is currently provided as part of a preview program pursuant to our [pre-release policies](/docs/licenses/license-information/referenced-policies/new-relic-pre-release-policy).
</Callout>

> This document covers the steps to troubleshoot common issues during the installation or operation of Agent Control.

Reviewer 2: ```suggestion
This document covers the steps to troubleshoot common issues when installing or operating Agent Control.
```

> To diagnose errors during the installation process, adjust the log level for Agent Control by adding the following setting in your values-newrelic.yaml file:

Reviewer 2: ```suggestion
## Enable debug logging
To diagnose errors during the installation process, you can increase the log level for Agent Control by adding the following setting in your `values-newrelic.yaml` file:
```

```yaml
agent-control-deployment:
  config:
    agentControl:
      content:  
        log:
          level: trace
```

> - **Default log level:** The default log level is `info`.

Reviewer 2: ```suggestion
- **Default log level:** `info`.
```

> - **Supported log levels:** Other supported log levels are `debug` and `trace`.

Reviewer 2: ```suggestion
- **Other supported log levels:** `debug` and `trace`.
```

> - **OTel collector debug logs:** To enable debug logs in the OpenTelemetry collector, add `verboseLog: true`.

Author: Please provide what the content should instead be.

> - **OTel collector debug logs:** To enable debug logs in the OpenTelemetry collector, add `verboseLog: true`.

Reviewer 2: ```suggestion
- **OTel collector logs:** To enable debug logs in the OpenTelemetry collector, add `verboseLog: true`.
```

> To read the logs, run the following command, making sure to adapt the command to reference the proper pod name and identifier:

Reviewer 2: ```suggestion
To inspect the Agent Control logs, run the following command, replacing `agent-control-***` with the name of your Agent Control pod:
```

```shell
kubectl logs agent-control-*** -n agent-control
```

## Status endpoint

> Agent Control exposes a local status endpoint. Used to understand the health of Agent Control and of managed agents. The endpoint is enabled by default on port `52100`. Follow these steps to query the cluster status:

Reviewer 2: ```suggestion
Agent Control exposes a local status endpoint you can use to check the health of Agent Control and its managed agents. This endpoint is enabled by default on port `52100`. Follow these steps to query the cluster status:
```

Forward a local port to the main agent-control pod:
```shell
kubectl port-forward <pod-name> 51200:51200
```
Request the agent status:
```shell
curl localhost:51200/status
```
## Helm release failure
Agent Control requires a valid authentication credential to securely connect to Fleet Control. Initially, this credential is automatically generated through the Agent Control installation UI and is represented by the `identityClientId` and `identityClientSecret` fields in the values file.

For security reasons, the credential necessary for installing Agent Control expires after 12 hours. If the installation fails with the error message below, follow these troubleshooting steps:
```shell
Error: UPGRADE FAILED: pre-upgrade hooks failed: job failed: BackoffLimitExceeded
```

> Check the logs of the Kubernetes job in charge of setting up the credential:

Reviewer 2: ```suggestion
Check the logs of the Kubernetes job responsible for setting up the Agent Control system identity.
```

First, identify the job’s pods:
```shell
kubectl describe job agent-control-system-identity -n <your-namespace>
```

> The Events section will point to the specific pods, as follows:

Reviewer 2: ```suggestion
In the Events section, look for entries for the specific pods, as follows:
```

```shell
Events:
      Type     Reason                Age   From            Message
      ----     ------                ----  ----            -------
      Normal   SuccessfulCreate      88s   job-controller  Created pod: agent-control-system-identity-installer-jr6cg
      Normal   SuccessfulCreate      73s   job-controller  Created pod: agent-control-system-identity-installer-wnx2v
      Normal   SuccessfulCreate      50s   job-controller  Created pod: agent-control-system-identity-installer-8zsqd
      Normal   SuccessfulCreate      7s    job-controller  Created pod: agent-control-system-identity-installer-btqh7
      Warning  BackoffLimitExceeded  1s    job-controller  Job has reached the specified backoff limit
```

> You can view the logs of the corresponding pods as follows:

Reviewer 2: ```suggestion
View the logs of the failing pods:
```

```shell
kubectl logs <pod-name> -n <your-namespace>
```

> Here's an example:

Reviewer 2: ```suggestion
Example:
```

```shell
kubectl logs agent-control-system-identity-installer-btqh7 -n newrelic
```

> Retry the installation via Helm and check the logs. Here're some known issues:

Reviewer 2: ```suggestion
After reviewing the logs, retry the installation using Helm while watching for specific error messages and checking the logs for potential problems. Below are some known issues and how to interpret them:
```

- **Invalid `identityClientId`:** `Error getting system identity auth token. The API endpoint returned 404: Failed to find Identity: <identityClientId-value>`
- **Invalid `identityClientSecret`:** `Error getting system identity auth token. The API endpoint returned 400: Bad client secret.`
- **Identity expired:** `Error getting system identity auth token. The API endpoint returned 400: Expired client secret.`

> - **Missing required permissions:** `Failed to create a New Relic System Identity for Fleet Control communication authentication. Please verify that your User Key is valid and that your Account Organization has the necessary permissions to create a System Identity: Exception while fetching data (/create) : Not authorized to perform this action or the entity is not found.`

Reviewer: The page looks weird

<img width="639" alt="Screenshot 2025-02-19 at 18 16 14" src="https://github.com/user-attachments/assets/e6bb5765-0039-4ec7-aa4f-9ec95dfaa94c" />

> - **Invalid `identityClientId`:** `Error getting system identity auth token. The API endpoint returned 404: Failed to find Identity: <identityClientId-value>`
> - **Invalid `identityClientSecret`:** `Error getting system identity auth token. The API endpoint returned 400: Bad client secret.`
> - **Identity expired:** `Error getting system identity auth token. The API endpoint returned 400: Expired client secret.`
> - **Missing required permissions:** `Failed to create a New Relic System Identity for Fleet Control communication authentication. Please verify that your User Key is valid and that your Account Organization has the necessary permissions to create a System Identity: Exception while fetching data (/create) : Not authorized to perform this action or the entity is not found.`

Reviewer 2: ```suggestion
- **Invalid identityClientId:**
  `Error getting system identity auth token. The API endpoint returned 404: Failed to find Identity: <identityClientId-value>`
- **Invalid identityClientSecret:**
  `Error getting system identity auth token. The API endpoint returned 400: Bad client secret.`
- **Identity expired:**
  `Error getting system identity auth token. The API endpoint returned 400: Expired client secret.`
- **Missing required permissions:**
  `Failed to create a New Relic System Identity for Fleet Control communication authentication. Please verify that your User Key is valid and that your Account Organization has the necessary permissions to create a System Identity: Exception while fetching data (/create) : Not authorized to perform this action or the entity is not found.`
```

## Invalid New Relic license

> If you see an error message in the OpenTelemetry collector deployment pod similar to the following, you may have set an invalid license key, causing the collector to be unable to export data to New Relic:

Reviewer 2: ```suggestion
If you see an error message like the one below in the logs of the OpenTelemetry collector deployment pod, it may indicate an invalid New Relic license key. This prevents the collector from being able to export telemetry data to New Relic:
```

```shell
2024-06-13T13:46:05.898Z error exporterhelper/retry_sender.go:126 Exporting failed. The error is not retryable. Dropping data. {"kind": "exporter", "data_type": "metrics", "name": "otlphttp/newrelic", "error": "Permanent error: error exporting items, request to https://otlp.nr-dat ││ go.opentelemetry.io/collector/exporter/exporterhelper.(*retrySender).send
```

### Solution

> Make sure you're using the correct license key.

Reviewer 2: ```suggestion
Confirm that you're using a valid New Relic license key in your configuration.
```

## OTel collector pods not created

> If you don't see pods related to the OTel collector being created, there's likely an issue with the HelmRelease Custom Resource Definition (CRD). Run the following command to diagnose the problem:

Reviewer 2: ```suggestion
If no OTel collector pods are being created, there may be an issue with the HelmRelease Custom Resource Definition (CRD).

 Check the status of the Helm release:
```

```shell
kubectl get helmrelease open-telemetry -n agent-control
```

> If everything went smoothly, you should see `READY` as `True`, and `STATUS` reflecting the current install succeeded.

Reviewer 2: ```suggestion
A successful and healthy release should show `READY: True` and `STATUS: InstallSucceeded`.
```

> If there are issues, the `STATUS` field will indicate the problem, and `READY` will be `FALSE`. Depending on the error, it might not be fully reflected in the status. For further insights, run the following command:

Reviewer 2: ```suggestion
If the release failed, the `STATUS` and `READY` fields will indicate the problem. Depending on the type of error, the root problem might not be fully reflected in the status field. To get more details, use `kubectl` to describe the HelmRelease resource:
```

```shell
kubectl describe helmrelease open-telemetry -n agent-control
```

## Additional support

> If you encounter issues not covered in this section, contact [New Relic support](https://support.newrelic.com).

Reviewer 2: ```suggestion
If you encounter issues not covered in this section, contact [New Relic support](https://support.newrelic.com) for further assistance.
```
